### ISSUE TITLE
Phase 1: 最初の分類器を実装する（TF-IDF + Logistic Regression）

### ISSUE THREAD

**Issue by yo4e** [yo4e/gaiden-sieve#2]
## 目的

Phase 0 で作った package / profile / data schema の上に、GAIDEN SIEVE 最初の分類器を実装する。

この Phase の主目的は高精度化ではなく、**機械学習の基本ループを実物で理解すること**。

```text
labeled data
    ↓
train / test split
    ↓
TF-IDF
    ↓
Logistic Regression
    ↓
prediction / probability
    ↓
Precision / Recall / F1
```

## 実装範囲

`DESIGN.md` の Phase 1 に従い、以下を実装する。

- [ ] `title + summary` を特徴量として使う
- [ ] TF-IDF でテキストを数値特徴量へ変換する
- [ ] Logistic Regression で二値分類する
- [ ] scikit-learn `Pipeline` を使い、学習時と推論時の前処理を同じ経路にする
- [ ] train / test split を実装する
- [ ] random seed を固定する
- [ ] Precision / Recall / F1 を計測する
- [ ] `predict_proba()` で `P(relevant)` を取得する
- [ ] profile の threshold を使い `relevant / uncertain / not_relevant` を返す
- [ ] ローカルで1件分類できる最小CLIを追加する
- [ ] pytest を追加・更新する
- [ ] README に Phase 1 の学習ポイントを短く追記する

## 学習時に確認すること

実装を進めながら、以下を山田さん向けに説明する。

1. **TF-IDF は何をしているのか**  
   文章をそのままモデルへ渡せないため、語の出現パターンを数値に変えている。

2. **Logistic Regression は何を学ぶのか**  
   TF-IDF の特徴量から relevant / not_relevant の境界を学び、確率を返す。

3. **なぜ train と test を分けるのか**  
   学習に使っていないデータで成績を見るため。既知問題の丸暗記を評価しない。

4. **Precision / Recall の違い**  
   AI外電では「関係ない記事を混ぜない」と「重要記事を取りこぼさない」のトレードオフを見る。

5. **なぜ `Pipeline` にするのか**  
   学習時と推論時で同じ TF-IDF 前処理を必ず使い、前処理ずれを防ぐ。

## 制約

- `label` / `label_reason` / `label_source` は特徴量に含めない（data leakage 防止）
- `uncertain` は学習クラスにしない。予測確率の中間帯として扱う
- RSS 接続はまだしない
- candidate artifact / quality gate / promotion は Phase 2 まで実装しない
- 高度なモデル、LLM、GPU、外部MLOps基盤へ広げない

## 完了条件

- [ ] `pytest` が通る
- [ ] fixture から学習できる
- [ ] test set に対する Precision / Recall / F1 が表示できる
- [ ] 新しい `title + summary` を1件渡して、確率と `relevant / uncertain / not_relevant` を返せる
- [ ] Phase 2 の model lifecycle へ進める状態になっている

## 正本

- `AGENTS.md`
- `DESIGN.md` §4, §6, §7, §8, §14, §17, §19
